### ISSUE TITLE
Clarification of "CREDENTIALS_TOKEN_C"

### ISSUE THREAD

**Issue by asyi23** [ocpi/ocpi#482]
I have a question , in credentials module , the TOKEN_A will be replace by TOKEN_C. What the reason behind it?

**Comment by andacata** [comment]
Security. The TOKEN_A is shared over a channel that you cannot guarantee is secure. If the TOKEN_A can only be used once, the risk is drastically reduced.

**Comment by asyi23** [comment]
> Security. The TOKEN_A is shared over a channel that you cannot guarantee is secure. If the TOKEN_A can only be used once, the risk is drastically reduced.

thank you for the answer. if u don't mind , i have another question regarding the contract id in token object . the emsp will create agreement with cpo . and then , emsp will create the contract id for a user. when a user of this emsp want to charging their car , the cpo will get the user contract id , cpo will know this is from which emsp based on the contract id. my concern is , how the cpo know this "contract id" belongs to this "emsp"? do cpo need to get updated with the list of contarct id that emsp own ?

**Comment by andacata** [comment]
Yes. The eMSP will update the CPO via Tokens Module, so the CPO must keep the relationship between the contract and the eMSP.

**Comment by cunhabarbosa** [comment]
About this topic, what is the format message when responding with TOKEN_C? I didn't find any example in the protocol and I don't know if this is the correct format.


```
{
    "data": {
          "token",
          "ZTA4MDdmOTUzYWQ5NDAwMDk0OWE0YzQ2MjUxYWNjNmE"
    },
    "status_code": 1000,
    "status_message": "Success",
    "timestamp": "2024-05-21T12:18:31Z"
}
```

Also, why is necessary to "Store CREDENTIALS_TOKEN_B" if only be used to get Available versions and endpoints?

**Comment by reinierl** [comment]
> About this topic, what is the format message when responding with TOKEN_C? I didn't find any example in the protocol and I don't know if this is the correct format.
> 
> ```
> {
>     "data": {
>           "token",
>           "ZTA4MDdmOTUzYWQ5NDAwMDk0OWE0YzQ2MjUxYWNjNmE"
>     },
>     "status_code": 1000,
>     "status_message": "Success",
>     "timestamp": "2024-05-21T12:18:31Z"
> }
> ```

That isn't even valid JSON so that's surely not the right format. 

There are example Credentials objects included in the specification though. You can use them with the response envelope to construct the right response containing Token C. You'd end up with something like:

```
{
     "data": {
        "token": "ebf3b399-779f-4497-9b9d-ac6ad3cc44d2",
        "url": "https://example.com/ocpi/versions",
        "roles": [{
            "role": "CPO",
            "party_id": "EXA",
            "country_code": "NL",
            "business_details": {
                "name": "Example Operator"
            }
        }]
    },
    "status_code": 1000,
    "status_message": "Success",
    "timestamp": "2024-05-21T12:18:31Z"
}
```

> Also, why is necessary to "Store CREDENTIALS_TOKEN_B" if only be used to get Available versions and endpoints?

Token B can be used for many things during the lifespan of the OCPI connection. It doesn't have to be used only for available versions and endpoints. If you do only use it for those purposes, you don't have to store it.
